### PR TITLE
[Recording] Counting feature in promptflow azure.

### PR DESCRIPTION
**Pull request by crazygao**
# Description

Counting feature in promptflow azure.

image

count value is the total count of all the other items.

-1, -2, -3 are error runs.
Some of the runs are Magic mocks, some of the runs cannot be found by pfazure runs get.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py [hunk 1]
```diff
@@ -624,6 +629,74 @@ def mock_trace_destination_to_cloud(subscription_id: str, resource_group_name: s
         yield
 
 
+@pytest.fixture(scope="class", autouse=is_live() and is_in_ci_pipeline())
+def counting_tokens_in_live(remote_client):
```
Comment (zhengfeiwang) [inline src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py:633]: may add some comments on this function, and highlight that this is for internal usage

File: src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py [hunk 2]
```diff
@@ -624,6 +629,74 @@ def mock_trace_destination_to_cloud(subscription_id: str, resource_group_name: s
...
+
+        from filelock import FileLock
+
+        count = sum(completed_run_metrics)
+        with FileLock(str(COUNTER_FILE) + ".lock"):
+            is_non_zero_file = os.path.isfile(COUNTER_FILE) and os.path.getsize(COUNTER_FILE) > 0
+            if is_non_zero_file:
+                with open(COUNTER_FILE, "r", encoding="utf-8") as f:
+                    number = json.load(f)
+                    number["count"] += count
```
Comment (zhengfeiwang) [inline src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py:692]: how about recording each run in each line? and we can apply aggregation in our collection pipeline?

**Review by zhengfeiwang — APPROVED** [review]

File: src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py (hunk 2 above)
Comment (crazygao) [reply src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py:692]: OK

File: src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py [hunk 3]
```diff
@@ -683,15 +687,16 @@ def mocked_run_method(self, *args, **kwargs):
...
         from filelock import FileLock
 
-        count = sum(completed_run_metrics)
+        count = sum(completed_run_metrics.values())
         with FileLock(str(COUNTER_FILE) + ".lock"):
             is_non_zero_file = os.path.isfile(COUNTER_FILE) and os.path.getsize(COUNTER_FILE) > 0
             if is_non_zero_file:
                 with open(COUNTER_FILE, "r", encoding="utf-8") as f:
                     number = json.load(f)
+                    number = number | completed_run_metrics
```
Comment (wangchao1230) [inline src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py:696]: why | here?

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py (hunk 3 above)
Comment (crazygao) [reply src/promptflow-azure/tests/sdk_cli_azure_test/conftest.py]: another way of joining two dicts I learnt from stackoverflow, but it doesn't seem to work properly.

**Review by zhengfeiwang — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**